R/learner_RWeka_classif_SMO.R

```diff
@@ -0,0 +1,156 @@
+#' @title Classification Support Vector Classifier Trained With John Platt's Sequential Minimal Optimization Algorithm
+#' @author damirpolat
+#' @name mlr_learners_classif.SMO
+#'
+#' @description
+#' Support Vector classifier trained with John Platt's sequential minimal optimization algorithm
+#' Calls [RWeka::SMO()] from \CRANpkg{RWeka}.
+#'
+#' @section Custom mlr3 parameters:
+#' - `output_debug_info`:
+#'   - original id: output-debug-info
+#'
+#' - `do_not_check_capabilities`:
+#'   - original id: do-not-check-capabilities
+#'
+#' - `num_decimal_places`:
+#'   - original id: num-decimal-places
+#'
+#' - `batch_size`:
+#'   - original id: batch-size
+#'
+#' - `E_poly`:
+#'   - original id: E
+#'
+#' - `L_poly`:
+#'   - original id: L
+#'
+#' - `C_poly`:
+#'   - original id: C
+#'
+#' - `C_logistic`:
+#'   - original id: C
+#'
+#' - `R_logistic`:
+#'   - original id: R
+#'
+#' - `M_logistic`:
+#'   - original id: M
+#'
+#' - Reason for change: This learner contains changed ids of the following control arguments
+#' since their ids contain irregular pattern
+#'
+#'
+#' @templateVar id classif.SMO
+#' @template learner
+#'
+#' @references
+#' `r format_bib(Platt1998, Keerthi2001, Hastie1998)`
+#'
+#' @template seealso_learner
+#' @template example
+#' @export
+LearnerClassifSMO = R6Class("LearnerClassifSMO",
+  inherit = LearnerClassif,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      param_set = ps(
+        subset = p_uty(tags = "train"),
+        na.action = p_uty(tags = "train"),
+        no_checks = p_lgl(default = FALSE, tags = "train"),
+        C = p_dbl(default = 1L, tags = "train"),
+        N = p_fct(default = "0", levels = c("0", "1", "2"), tags = "train"),
+        L = p_dbl(default = 1.0e-3, tags = "train"),
+        P = p_dbl(default = 1.0e-12, tags = "train"),
+        M = p_lgl(default = FALSE, tags = "train"),
+        V = p_int(default = -1L, tags = "train"),
+        W = p_int(default = 1L, tags = "train"),
+        K = p_uty(default = "weka.classifiers.functions.supportVector.PolyKernel", tags = "train"),
```

Comment:
There is a mismatch with the tuning spaces. You called the kernel `"NormalizedPolyKernel", "PolyKernel", "Puk", "RBFKernel"` there.

Reply:
The dependency below have the prefix `weka.classifiers.functions.supportVector`

Reply:
Yes, RWeka can do both longer and shorter names. Let me shorten the names to make it consistent across mlr.

Reply:
I shortened all values. Some params are not fully shortened like `E` from `BayesNet` learner because the learner cannot recognize short names so I added prefix to them like `estimate.SimpleEstimator`.